File: front_end/lib/app/core/models/result.dart
class Result {
  final int id;
  final int candidateId;
  final int questionId;
  final List<String> chosenAnswers;
  final bool correct;

  Result({
    this.id = -1,
    this.candidateId = -1,
    required this.questionId,
    required this.chosenAnswers,
    required this.correct,
  });

  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'questionId': questionId,
      'chosenAnswers': chosenAnswers,
      'correct': correct,
    };
  }

  factory Result.fromJson(Map<String, dynamic> map) {
    return Result(
      id: map['id'] as int,
      candidateId: map['candidateId'] as int,
      questionId: map['questionId'] as int,
      chosenAnswers:
          (map['chosenAnswers'] as List).map((e) => e.toString()).toList(),
      correct: map['correct'] as bool,
    );
  }
}

File: microservice/result-service/src/main/java/com/example/resultservice/entity/Result.java
package com.example.resultservice.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;

import java.util.List;

@Entity
@Table(name = "result")
public class Result {
    @Id
    @Column(name = "id", length = 10)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private boolean correct;
    @ManyToOne(cascade = CascadeType.MERGE, fetch= FetchType.LAZY)
    @JoinColumn(name = "candidate_id")
    @JsonBackReference
    private Candidate candidate;

    @Column(name = "question_id")
    private int questionId;

    @ElementCollection
    @CollectionTable(name = "chosen_answers", joinColumns = @JoinColumn(name = "result_id"))
    @Column(name = "chosen_answer")
    private List<String> chosenAnswers;


    public Result() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public boolean isCorrect() {
        return this.correct;
    }

    public void setCorrect(boolean correct) {
        this.correct = correct;
    }

    public Candidate getCandidate() {
        return candidate;
    }

    public void setCandidate(Candidate candidate) {
        this.candidate = candidate;
    }


    public List<String> getChosenAnswers() {
        return chosenAnswers;
    }

    public void setChosenAnswers(List<String> chosenAnswers) {
        this.chosenAnswers = chosenAnswers;
    }

    public int getQuestionId() {
        return questionId;
    }

    public void setQuestionId(int questionId) {
        this.questionId = questionId;
    }
}

